mbuild/conversion.py

```diff
@@ -0,0 +1,1364 @@
+    coords_only : bool, optional, default=False
+        Set preexisting atoms in compound to coordinates
+        given by structure.
+    infer_hierarchy : bool , optional, default=True
+        If True, infer compound hierarchy from chain
+        and reisdues
+
+    Returns
+    -------
+    compound : mb.Compound
+
+    """
+    # Check coords_only option
+    if compound and coords_only:
+        if len(structure.atoms) != compound.n_particles:
+            raise ValueError(
+                'Number of atoms in {structure} does not '
+                '{compound}'.formats(**locals()))
+        atoms_particles = zip(structure.atoms,
+                    compound.particles(include_ports=False))
+        if None in compound._particles(include_ports=False):
+            raise ValueError('Some particles are None')
+        for pmd_atom, particle in atoms_particles:
+            particle.pos = np.array([pmd_atom.xx,
+                                     pmd_atom.xy,
+                                     pmd_atom.xz]) / 10
+        return compound
+    elif not compound and coords_only:
+        raise MBuildError('coords_only=True but'
+                'host compound is not provided')
+
+    # Initialize a compound if none is provided
+    if not compound:
+        compound = mb.Compound()
+
+    # Convert parmed structure to mbuild compound
+    atom_mapping = dict()
+    chain_id = None
+    chains = defaultdict(list)
+
+    # Build up chains dict
+    # ?Could I change this to normal dict?
+    for residue in structure.residues:
+        chains[residue.chain].append(residue)
+
+    # Build up compound
+    for chain, residues in chains.items():
+        if len(chain) > 1:
+            chain_compound = mb.Compound()
+            compound.add(chain_compound, chain_id)
+        else:
+            chain_compound = compound
+        for residue in residues:
+            if infer_hierarchy:
+                residue_compound = mb.Compound(name=residue.name)
+                chain_compound.add(residue_compound)
+                parent_compound = residue_compound
+            else:
+                parent_compound = chain_compound
+            for atom in residue.atoms:
+                pos = np.array([atom.xx,
+                                atom.xy,
+                                atom.xz]) / 10
+                new_atom = mb.Particle(name=str(atom.name),
+                                    pos=pos)
+                parent_compound.add(new_atom, label='{0}[$]'.format(atom.name))
+                atom_mapping[atom] = new_atom
+
+    # Infer bonds information
+    for bond in structure.bonds:
+        atom1 = atom_mapping[bond.atom1]
+        atom2 = atom_mapping[bond.atom2]
+        compound.add_bond((atom1, atom2))
+
+    # Convert box information
+    if structure.box is not None:
+        compound.periodicity = structure.box[0:3] / 10
+    else:
+        compound.periodicity = np.array([0., 0., 0.])
```

Comment:
What about any off-diagonal elements of the box? Should we throw a warning at least?

Reply:
Sorry, I am not entirely sure what you mean, can you elaborate a bit? Do you mean adding a warning for when we make the periodicity = [0, 0, 0]?

Reply:
I mean the parmed `Structure.box` (https://parmed.github.io/ParmEd/html/structobj/parmed.structure.Structure.html) is a list of 6 floats, `a,b,c, alpha, beta, gamma`. You are throwing away `alpha, beta, gamma`. If those are not 90 degrees, that is important information. I think you should at the least throw a warning if that happens.

Reply:
I see, I will throw in a warning for that! Thanks!